File: leet_1759_count_substring.cc
class Solution {
public:
    int countHomogenous(string s)
    {
        if (s == "")
            return 0;
        long long MD = 1e9 + 7;
        long long res = 0;
        char c = s[0];
        int len = s.size();
        int dis = 1;
        int i = 0;
        while (++i < len)
        {
            if (c == s[i])
            {
                dis++;
            }
            else
            {
                res += ((dis % MD + 1) * dis % MD / 2) % MD;
                c = s[i];
                dis = 1;
            }
        }
        return res + ((dis % MD + 1) * dis % MD / 2) % MD;
        // return res;
    }
};

File: leet_1759_count_substring.py
class Solution:
    def countHomogenous(self, s: str) -> int:
        MD = int(1e9 + 7)
        res = 0
        dis = 1
        c = s[0]
        for i in range(1, len(s)):
            if c == s[i]:
                dis += 1
            else:
                res += ((dis * (dis + 1)) // 2) % MD
                dis = 1
                c = s[i]
        res += ((dis * (dis + 1)) // 2) % MD
        return res

